common/djangoapps/student/models.py

```diff
@@ -1847,3 +1847,49 @@ class CourseEnrollmentAttribute(models.Model):
         max_length=255,
         help_text=_("Value of the enrollment attribute e.g. ASU")
     )
+
+    @classmethod
+    def add_or_update_enrollment_attr(cls, enrollment, value, namespace, name):
+        """Add 'CourseEnrollmentAttribute' for given enrollment if not exist
+        else update.
+
+        Args:
+            enrollment(CourseEnrollment): 'CourseEnrollment' for which attribute is to be added
+            value(str): Value of the enrollment attribute
+            namespace(str): Namespace of the enrollment attribute
+            name(str): Name of the enrollment attribute
+        """
+        enrollment, created = cls.objects.get_or_create(
+            enrollment=enrollment, namespace=namespace, name=name, defaults={'value': value}
+        )
+        if not created:
+            enrollment.value = value
+            enrollment.save()
+
```

Comment:
I still think we want to do this in two steps:
1. Delete all existing attributes.
2. Bulk create the specified attributes.

This has a number of advantages:
1. It ensures that the number of database queries is constant with respect to the number of attributes.
2. It allows the caller to remove attributes (for example, when unenrolling a user from a credit seat).
3. It allows the caller to specify exactly what attributes should be present without needing to worry about the current state of the attributes.